Realm/RLMObject.h

```diff
@@ -322,7 +332,21 @@ RLM_ASSUME_NONNULL_BEGIN
  @return    An object of the subclass type or nil if an object with the given primary key does not exist.
  @see       -primaryKey
  */
-+ (nullable instancetype)objectForPrimaryKey:(nullable id)primaryKey;
++ (nullable instancetype)objectForPrimaryKey:(nullable id)primaryKey DEPRECATED_MSG_ATTRIBUTE("use +[RLMObject objectForObjectID:]");
+
+/**
+ Get the single object with the given object ID from the default Realm.
+
+ Returns the object from the default Realm which has the given object ID, or
+ `nil` if the object does not exist. This is slightly faster than the otherwise
+ equivalent `[[SubclassName objectsWhere:@"objectIDPropertyName = %@", key] firstObject]`.
+
+ This method requires that `objectID` be overridden on the receiving subclass.
+
+ @return    An object of the subclass type or nil if an object with the given object ID does not exist.
+ @see       -objectID
+ */
++ (nullable instancetype)objectForObjectID:(nullable id)objectID;
```

Comment:
I think including `ObjectID` in the name is redundant given object is already mentioned at the start of the name.

Is *for* the best word to use? `objectForID` sounds awkward to me. `objectWithID` sounds more natural.